src/Symfony/Component/OptionsResolver/OptionsResolver.php

```diff
@@ -987,58 +987,43 @@ public function offsetGet($option/*, bool $triggerDeprecation = true*/)
         return $value;
     }
 
-    private function verifyTypes(string $type, $value, array &$invalidTypes): bool
+    private function verifyTypes(string $type, $value, array &$invalidTypes, int $level = 0): bool
     {
         if (\is_array($value) && '[]' === substr($type, -2)) {
-            return $this->verifyArrayType($type, $value, $invalidTypes);
-        }
-
-        if (self::isValueValidType($type, $value)) {
-            return true;
-        }
-
-        if (!$invalidTypes) {
-            $invalidTypes[$this->formatTypeOf($value, null)] = true;
-        }
-
-        return false;
-    }
-
-    private function verifyArrayType(string $type, array $value, array &$invalidTypes, int $level = 0): bool
-    {
-        $type = substr($type, 0, -2);
-
-        $suffix = '[]';
-        while (\strlen($suffix) <= $level * 2) {
-            $suffix .= '[]';
-        }
-
-        if ('[]' === substr($type, -2)) {
-            $success = true;
-            foreach ($value as $item) {
-                if (!\is_array($item)) {
-                    $invalidTypes[$this->formatTypeOf($item, null).$suffix] = true;
+            $type = substr($type, 0, -2);
 
+            foreach ($value as $val) {
+                if (!$this->verifyTypes($type, $val, $invalidTypes, $level + 1)) {
                     return false;
                 }
-
-                if (!$this->verifyArrayType($type, $item, $invalidTypes, $level + 1)) {
-                    $success = false;
-                }
             }
 
-            return $success;
+            return true;
         }
 
-        foreach ($value as $item) {
-            if (!self::isValueValidType($type, $item)) {
-                $invalidTypes[$this->formatTypeOf($item, $type).$suffix] = $value;
+        try {
+            if ('null' === $type) {
+                if (null === $value) {
+                    return true;
+                }
+            } elseif (('is_'.$type)($value)) {
+                return true;
+            }
+        } catch (\Error $e) {
```

Comment:
Swallowing any `Error` is very suspicious to me. can we be more specific here? Shouldn't we rethrow? The previous had not such try/catch.

Reply:
I thought about removing all `\function_exists()` calls, but you're right, we could be hiding some real errors from custom `is_*` functions. 

Rolling back to the `\function_exists()` version.